state/utils/ttl.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+)
+
+// Key used for "ttlInSeconds" in metadata.
+const MetadataTTLKey = "ttlInSeconds"
```

Comment:
package contrib `metadata` `TryGetTTL` already does all of this. If necessary just fix up that method instead please.

Reply:
Prefer using https://github.com/dapr/components-contrib/blob/master/metadata/utils.go#L58 `TryGetTTL`

Reply:
This method can't be used because it rejects values < 0. For TTLs in state store set requests, we support -1 to indicate to not set a TTL. This is an explicit contract in our docs: https://docs.dapr.io/developing-applications/building-blocks/state-management/state-store-ttl/#persisting-state-ignoring-an-existing-ttl

For that reason, fixing-up the method in contrib metadata is not an option IMHO.

Reply:
Aren't there other components supporting TTL? If so, let's at least move your code into the metadata package so it can be found again.

Reply:
> Aren't there other components supporting TTL?

What do you mean?

I put it in `state/utils` because it's specific to state stores.